api/src/opentrons/protocol_api/validation.py

```diff
@@ -180,3 +196,63 @@ def ensure_valid_labware_offset_vector(
     if not all(isinstance(v, (float, int)) for v in offsets):
         raise TypeError("Offset values should be a number (int or float).")
     return offsets
+
+
+class WellTarget(NamedTuple):
+    """A movement target that is a well."""
+
+    well: Well
+    location: Optional[Location]
+    in_place: bool
+
+
+class PointTarget(NamedTuple):
+    """A movement to coordinates"""
+
+    location: Location
+    in_place: bool
+
+
+class NoLocationError(ValueError):
+    """Error representing that no location was supplied."""
+
+
+class LocationTypeError(TypeError):
+    """Error representing that the location supplied is of different expected type."""
+
+
+def validate_location(
+    location: Union[Location, Well, None], last_location: Optional[Location]
+) -> Union[WellTarget, PointTarget]:
+    """Validate and return if should use a WellTarget or a PointTarget.
+    Args:
+        location: The input location.
+        last_location: The cached last location.
+    Raises:
+        NoLocationError: The is no input location and no cached loaction.
+        LocationTypeError: The location supplied is of unexpected type.
+    """
+    from .labware import Well
+
+    target_location = location or last_location
+
+    if target_location is None:
+        raise NoLocationError()
+
+    if not isinstance(target_location, (Location, Well)):
+        raise LocationTypeError(
+            "location should be a Well or Location, but it is {}".format(location)
```

Comment:
Always prefer f-strings

```suggestion
            f"location should be a Well or Location, but it is {location}"```